RA-03-create-buttons

DenisGubernatorov opened the pull request:
1) Подключен ViewBinding 
2)В методе onCreate() инициализируется binding и передается в setContentView
3) В основной контейнер добавлен контейнер для кнопок с настройками отступа. Значения отступов берутся из ресурсного фай
4) В ресурсы добавлена иконка для кнопки "Избранное"
5) В ресурсы добавлен шрифт montserrat_alternates_semibold 
6) Реализованы кнопки с настройкой стиля кнопок и текста из ресурсных файлов стилей.
7) Для StatusBar и NavigationBar фон сделан как и основной

DenisGubernatorov committed:
RA-03-create-project
1) Переименование файла иконки приложения согласно правилам именования

DenisGubernatorov committed:
RA-03-create-project
1) Подключен ViewBinding
2)В методе onCreate() инициализируется binding и передается в setContentView
3) В основной контейнер добавлен контейнер для кнопок с настройками отступа. Значения отступов берутся из ресурсного фай
4) В ресурсы добавлена иконка для кнопки "Избранное"
5) В ресурсы добавлен шрифт montserrat_alternates_semibold
6) Реализованы кнопки с настройкой стиля кнопок и текста из ресурсных файлов стилей.
7) Для StatusBar и NavigationBar фон сделан как и основной

aGroupoid commented on app/src/main/java/com/example/recipesapp/MainActivity.kt:
```diff
@@ -3,18 +3,18 @@ package com.example.recipesapp
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        _binding = ActivityMainBinding.inflate(layoutInflater)
         enableEdgeToEdge()
```
Давай уберем `enableEdgeToEdge()`, чтобы контент активити (и фрагмента) не заезжал на статусбар и навбар.

aGroupoid commented on app/src/main/java/com/example/recipesapp/MainActivity.kt:
```diff
@@ -3,18 +3,18 @@ package com.example.recipesapp
 
 class MainActivity : AppCompatActivity() {
+    private var _binding: ActivityMainBinding? = null
+    private val binding
```
В уроке реализовывался backing property (резервное свойство). Но здесь предлагаю применить более элегантный вариант с помощью делегата lazy, который использует ленивую инициализацию. Это такой механизм, который в лямбде принимает в нашем случае inflate() и возвращает созданный объект, который будет создан при первом обращении к нему.

aGroupoid commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -5,15 +5,45 @@
-        app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+        android:layout_marginStart="@dimen/margin_full_16_dp"
+        android:layout_marginEnd="@dimen/margin_full_16_dp"
```
Не обязательно для каждой стороны прописывать марджины по отдельности. Есть более универсальные атрибуты.
android:layout_margin=""
android:layout_marginHorizontal=""
android:layout_marginVertical=""

В твоем случае можно заменить на android:layout_marginHorizontal=""

aGroupoid commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -5,15 +5,45 @@
+            style="@style/StyleNavButton"
+            android:layout_width="0dp"
+            android:layout_height="wrap_content"
+            android:layout_marginEnd="4dp"
```
Все значения не забываем выносить в ресурсы и переиспользовать. Размеры элементов/шрифтов, цвета, строки

aGroupoid commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,3 +1,5 @@
 <resources>
     <string name="app_name">RecipesApp</string>
+    <string name="category">Категории</string>
+    <string name="favorites">Избранное</string>
```
Чтобы при глобальном поиске по проекту (через Shift + Ctrl + F) в выдачу не попадали другие результаты, содержащие этот же текст, и не сбивали с толку,
лучше сразу приучаться к уточнению, что это текстовый ресурс с помощью префикса, например, text_ или title_.

aGroupoid commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,18 @@
+        <item name="android:fontFamily">@font/montserrat_alternates_semibold</item>
+        <item name="android:textSize">14sp</item>
+        <item name="android:textAllCaps">true</item>
+        <item name="android:textColor">#FFFFFF</item>
```
Все значения не забываем выносить в ресурсы и переиспользовать. Размеры элементов/шрифтов, цвета, строки.

Это касается и элеметнов стилей

aGroupoid commented on app/src/main/res/values/themes.xml:
```diff
@@ -1,8 +1,11 @@
     <!-- Base application theme. -->
     <style name="Base.Theme.RecipesApp" parent="Theme.Material3.DayNight.NoActionBar">
         <!-- Customize your light theme here. -->
         <!-- <item name="colorPrimary">@color/my_light_primary</item> -->
```
От этих информационных комментов можне уже избавиться

aGroupoid commented:
Нейминги коммита, пулл реквеста и ветки следует делать все в едином стиле (который принят в сообществе). Как правило это аббревиатура название проекта.

Например:
RA-03 create navigation buttons (для названия PR и сообщения коммита)
RA-03-create-navigation-buttons (для названия ветки, разделение всех слов дефисами)

Давай сразу следовать этой договоренности. Щас немного не соответсвует название коммитов и пулл реквеста

DenisGubernatorov committed:
RA-03-create-project
1) com.example.recipesapp.MainActivity.getBinding переведен на ленивую инициализацию
2) Абсолютные значения в атрибутах элементов переведены на использование ресурсных файлов

aGroupoid commented on app/src/main/res/values/strings.xml:
```diff
@@ -1,5 +1,5 @@
-    <string name="category">Категории</string>
-    <string name="favorites">Избранное</string>
+    <string name="title_app_name">RecipesApp</string>
+    <string name="btnTxt_category">Категории</string>
```
лучше не конкретизировать, в каком элементе используется та или иная надпись. text_ достаточно, чтоб отличить по быстрому поиску, что это текст. Надпись Категория может быть не только на кнопке, просто в каком-нибудь из textView

aGroupoid reviewed (approved):
Всё верно, молодец! Оставил небольшой коммент, поправь его уже в следующем задании

На будущее, предлагаю тебе не резолвить треды обсуждений здесь, а то мне все равно приходится все раскрывать при повторной проверке, чтобы входить в контекст)
